presto-docs/src/main/sphinx/develop/group-provider.rst

```diff
@@ -0,0 +1,36 @@
+==============
+Group provider
+==============
+
+Presto can map usernames onto groups for easier access control management.
+This mapping is performed by a ``GroupProvider`` implementation.
+
+Implementation
+--------------
+
+``GroupProviderFactory`` is responsible for creating a ``GroupProvider`` instance.
+It also defines the name of the group provider as used in the configuration file.
+
+``GroupProvider`` contains a one method, ``getGroups(String user)`` which returns a ``Set<String>`` of group names.
+This set of group names becomes part of the ``Identity`` and ``ConnectorIdentity`` objects representing the user, and can then be used by :doc:`system-access-control`.
```

Comment:
Nit: we try to wrap documentation at 80-100 characters for readability. (this is not a hard rule ... if it's just one word or two, don't wrap, as that's harder to read)